File: Milktea/LeetCode/Code/JavaScript/757.js
const intersectionSizeTwo = (intervals) => {
    intervals.sort((a, b) => {
        if (a[1] === b[1]) {
            return b[0] - a[0];
        }
        return a[1] - b[1];
    });

    let setSize = 0;
    let small = -1;
    let big = -1;

    for (let i = 0; i < intervals.length; i++) {
        const [l, r] = intervals[i];
        if (l > big) {
            small = r - 1;
            big = r;
            setSize += 2;
        } else if (l > small) {
            small = big;
            big = r;
            setSize += 1;
        }
    }

    return setSize;
}   

File: Milktea/LeetCode/Code/Python/757.py
class Solution:
    def intersectionSizeTwo(self, intervals: List[List[int]]) -> int:
        intervals.sort(key = lambda x: (x[1], -x[0]))

        setSize = 0
        small, big = -1, -1

        for l, r in intervals:
            if l > big:
                small = r - 1
                big = r
                setSize += 2
            elif l > small:
                small = big
                big = r
                setSize += 1
        
        return setSize